keps/sig-node/3573-device-plugin/README.md

```diff
@@ -493,12 +614,195 @@ Follow protobuf guidelines on versioning:
+
+Even when this feature is enabled, cluster admin needs to provision device plugins
+(typically deployed as a daemonset) on the cluster which subsequently be requested
+by the users. 
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, covered by node e2e tests:
+* https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/device_manager_test.go
+* https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/device_plugin_test.go
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+* A rollback to Kubernetes version <1.10 can impact running workloads as they require
+  API version to exactly match for Kubelet and Device Plugin.
+
+* A rollout can fail in case a bug is introduced either in device manager or device
+  plugin preventing already running pods from restarting or new pods to start.
+
+###### What specific metrics should inform a rollback?
+
+Not Applicable, metrics specific to rollback are not available.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
+Yes
+The device plugin API is stable since its graduation to Beta and device plugins using
+`v1beta1` API are compatible with any Kuberntes version >= 1.10. Therefore, ugrades or
+downgrades don't have an impact as long as the upgraded/downgraded version is > Kubernetes
+v1.10.
+
+When upgrading device plugins, it is recommended that the cluster admin drain workloads
+requesting devices and conform to the device plugin API.
+
+This has been validated by several device plugins. Here is an example: https://github.com/NVIDIA/k8s-device-plugin#upgrading-kubernetes-with-the-device-plugin. 
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+1. Kubelet PodResource API endpoints (`List` and `GetAllocatableResources`) can be used
+   to obtain information on devices allocated to running pods. Refer to
+   [keps/sig-node/606-compute-device-assignment]([keps/sig-node/606-compute-device-assignment]).
+2. Status of the nodes  specifically device `capacity` and `allocatable` can be inspected
+   to see if device plugins were successflly deployed and are avaialable to be allocated
+   on the node.
+   In addition to that, status of the running pods can be inspected to determine if the
+   pod requesting device(s) was running successfully and allocated device(s) as per its
+   request.
+3. Typically, device allocation to a workload is accompanied by creation of environment
+   variable capturing device specification identification information e.g. PCI address
+   information which can be inspected, creation of device files and device specific
+   volume mount points.This varies depending on the device plugin implementation. 
```

Comment:
IIUC from what I get from [the review of a similar case](https://github.com/kubernetes/enhancements/pull/3350/files/9ed4fd204ea3952f84535036e20f15e5d1c2ec9b#r985978216), if we need to tap inside pods, than this is not a good monitoring strategy. But let's doublecheck this :)

Reply:
Thanks for pointing this out, looks like this is going to be a gap for both Device Manager and CPU Manager. Let's dig into it and see if we have metrics (that we are not aware of) and evaluate the effort involved in adding them. I wonder if it is acceptable to proceed with GA graduation if we plan to add metric(s) in the same cycle as part of the development phase before code freeze or we need to give it time for it to get baked in.

Reply:
It appears we have device plugin related metrics [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L63-L64). Thanks @fromani for identifying and pointing me this!